src/Symfony/Component/Translation/Dumper/FileDumper.php

```diff
@@ -82,12 +82,27 @@ public function dump(MessageCatalogue $messages, $options = array())
                 }
             }
             // save file
-            file_put_contents($fullpath, $this->format($messages, $domain));
+            file_put_contents($fullpath, $this->formatCatalogue($messages, $domain, $options));
         }
     }
 
     /**
      * Transforms a domain of a message catalogue to its string representation.
+     * This function provided for backward compatibility.
```

Comment:
>array should be string[]
sorry, but why? it's array options like array('as_tree' => true), not an array of strings.

done your other notes.